traffic_ops/traffic_ops_golang/cdnfederation/cdnfederations.go

```diff
@@ -0,0 +1,414 @@
+
+	if rowsAffected != 1 {
+		if rowsAffected < 1 {
+			return errors.New("no federation found with this id"), tc.DataMissingError
+		}
+		return fmt.Errorf("this update affected too many rows: %d", rowsAffected), tc.SystemError
+	}
+
+	return nil, tc.NoError
+}
+
+// In the perl version, :name is ignored. It is not even verified whether or not
+// :name is a real cdn that exists. This mimicks the perl behavior.
+func (fed *TOCDNFederation) Delete() (error, tc.ApiErrorType) {
+
+	if ok, err := fed.isTenantAuthorized(); err != nil {
+		log.Errorf("checking tenacy: %v", err)
+		return tc.DBError, tc.SystemError
+	} else if !ok {
+		return tc.TenantUserNotAuthError, tc.ForbiddenError
+	}
+
+	log.Debugf("about to run exec query: %s with federation: %++v", deleteQuery(), fed)
+	result, err := fed.ReqInfo.Tx.NamedExec(deleteQuery(), fed)
+	if err != nil {
+		log.Errorf("received error: %++v from delete execution", err)
+		return tc.DBError, tc.SystemError
+	}
+
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		log.Errorf("getting rows affected: %v", err)
+		return tc.DBError, tc.SystemError
+	}
+
+	if rowsAffected != 1 {
+		if rowsAffected < 1 {
+			return errors.New("no federation with that id found"), tc.DataMissingError
+		}
+		return fmt.Errorf("this delete affected too many rows: %d", rowsAffected), tc.SystemError
+	}
+	return nil, tc.NoError
+}
+
+// Function not exported because although DELETE and UPDATE have normal tenacy check,
+// CREATE does not. No ds is associated on create. This isn't used for READ because
+// psql doesn't like nested queries within the same transaction.
+func (fed TOCDNFederation) isTenantAuthorized() (bool, error) {
+
+	tenantID, err := getTenantIDFromFedID(*fed.ID, fed.ReqInfo.Tx.Tx)
+	if err != nil {
+		// If nobody has claimed a tenant, that federation is publicly visible.
+		// This logically follows /federations/:id/deliveryservices
+		if err == sql.ErrNoRows {
+			return true, nil
+		}
+		log.Errorf("getting tenant id from federation: %v", err)
+		return false, err
+	}
+
+	// TODO: After IsResourceAuthorizedToUserTx is updated to no longer have `use_tenancy`,
+	// that will probably be better to use. For now, use the list. Issue #2602
+	list, err := tenant.GetUserTenantIDListTx(fed.ReqInfo.Tx.Tx, fed.ReqInfo.User.TenantID)
+	if err != nil {
+		return false, err
+	}
+	for _, id := range list {
+		if id == tenantID {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func getTenantIDFromFedID(id int, tx *sql.Tx) (int, error) {
+	tenantID := 0
+	query := `
+	SELECT ds.tenant_id from federation as f
+	JOIN federation_deliveryservice as fd ON f.id = fd.federation
+	JOIN deliveryservice as ds ON ds.id = fd.deliveryservice
+	WHERE f.id = $1`
+	err := tx.QueryRow(query, id).Scan(&tenantID)
+	return tenantID, err
+}
+
+func selectByID() string {
+	return `
+	SELECT tenant_id, federation.id as id, cname, ttl, description, federation.last_updated as last_updated, ds.id as ds_id, xml_id FROM federation
```

Comment:
can you split up each column in the `SELECT` part onto their own lines? And put the `FROM` on its own line as well? Also, please specify which table the column is from (e.g. federation.cname, etc).